pkg/ddl/ddl_api.go

```diff
@@ -1491,7 +1491,7 @@ func getDefaultValue(ctx sessionctx.Context, col *table.Column, option *ast.Colu
 			return str, false, err
 		}
 		// For other kind of fields (e.g. INT), we supply its integer as string value.
-		value, err := v.GetBinaryLiteral().ToInt(ctx.GetSessionVars().StmtCtx.TypeCtx())
+		value, err := v.GetBinaryLiteral().ToInt(ctx.GetEvalCtx().TypeCtx())
```

Comment:
Because the type of input context `getDefaultValue` is changed from `sessionctx.Context` to `BuildContext` to reduce context requirement. We can only get the `types.Context` from `ctx.GetEvalCtx()`